apps/ordermoments/order_moments.star

```diff
@@ -0,0 +1,749 @@
+"""
+Applet: Order Moments
+Summary: Celebrate major moments
+Description: Get celebratory notifications when you hit specific order milestones.
+Author: Shopify
+"""
+
+load("render.star", "render")
+load("schema.star", "schema")
+load("http.star", "http")
+load("time.star", "time")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("humanize.star", "humanize")
+load("cache.star", "cache")
+load("hash.star", "hash")
+
+# Messages
+ERROR_TEXT = "We hit a snag. Please check your app."
+
+# Endpoints
+REST_ENDPOINT = "https://{}.myshopify.com/admin/api/2022-10/{}.json"
+COUNT_ENDPOINT = "orders/count"
+
+# Metafield definitions
+METAFIELD_ENDPOINT = "metafields"
+METAFIELD_UPDATE_ENDPOINT = "metafields/{}"
+METAFIELD_DESCRIPTION = "A number of sales last celebrated by this store's TidByt Display"
+METAFIELD_NAMESPACE = "tidbyt"
+METAFIELD_KEY = "lastcelebration"
+METAFIELD_OWNER = "shop"
+
+# Cache definitions
+CACHE_TTL = 600
+CACHE_ID_ORDERS = "{}-orders"
+CACHE_ID_METAFIELD = "{}-metafields"
+
+# Milestone definitions are a list of Tuples where the first element is a base number of sales
+# and the second element is an increment at which celebrations happen after that base.
+MILESTONE_DEFINITIONS = [
+    (0, 10),
+    (100, 25),
+    (500, 50),
+    (1000, 100),
+    (2000, 250),
+    (5000, 500),
+    (10000, 1000),
+    (100000, 5000),
+    (500000, 10000),
+    (1000000, 100000),
+]
+
+# MAIN
+# ----
+def main(config):
```

Comment:
One thing to note is that we will call this app to render previews inside of the Tidbyt Mobile app. You should consider adding some default view for a preview so that users browsing on the explore tab or using search can find a representative preview

Reply:
Good point! Is there a config value that's passed to the app for preview rendering?